### PR TITLE
feat(wellinks-extension): adds an insertMemberListEvent action

### PR DESCRIPTION
**Pull request by AshvinP-S**
Creates an action in the Wellinks Extension to hit a Lambda in one of our SAM projects. As it requires an API Key to use, the proper settings have been added to the extension.

### CODE REVIEW

**Comment by AshvinP-S** [comment]
Added WIP to PR till the deploy of the Wellinks Platform SAM

File: extensions/wellinks/actions/insertMemberListEvent/insertMemberListEvent.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      if (isNil(settings.platformApiUrl) || isNil(settings.platformApiKey)) {
+        throw new Error(
+          'The Platform API URL and/or API Key is not set in the settings'
+        )
+      }
+      const client = new WellinksClient(
+        settings.platformApiUrl,
+        settings.platformApiKey
+      )
+      if (!isNil(client)) {
```
Comment (bejoinka) [inline extensions/wellinks/actions/insertMemberListEvent/insertMemberListEvent.ts:99]: heya, there's no `else` here, which leaves a path where neither onComplete nor onError are called, causing the activity to hang forever. Given the client is created just above, you're safe removing this if statement.

File: extensions/wellinks/wellinksClient.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export class WellinksClient {
+  private readonly _apiUrl: string
+  private readonly _apiKey: string
+  constructor(apiUrl: string, apiKey: string) {
+    this._apiKey = apiKey
+    this._apiUrl = apiUrl
+  }
+
+  readonly memberListEvent = {
+    insert: async (
```
Comment (bejoinka) [inline extensions/wellinks/wellinksClient.ts:12]: totally preference, but when dealing with more than two or three input parameters, i tend to like to use a de-structured object as input. e.g.
```ts
insert: async ({
  foo,
  bar,
  baz,
}: {
  foo: string
  bar: string
  baz: string
}): ...
```

That pattern will help with autocompletion and code maintenance down the road

File: extensions/wellinks/actions/insertMemberListEvent/insertMemberListEvent.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+          isNil(sendgridListId) ||
+          isNil(originatorName) ||
+          isNil(eventDate)
+        ) {
+          await onError({
+            events: [
+              {
+                date: new Date().toISOString(),
+                text: {
+                  en: 'Some or all of the arguments are missing.',
```
Comment (bejoinka) [inline extensions/wellinks/actions/insertMemberListEvent/insertMemberListEvent.ts:113]: FWIW Rob or other builders won't be able to figure out which arguments are missing unless you put them in the error.

**Review by bejoinka — APPROVED** [review]
heya, i'll approve and send along to unblock you and monica.

let's circle up on a few things with this extension when we move it private:
1. using zod for validation
2. error messaging
3. using settings

i'd ultimately love for you guys to be featured as a best practice in your third-party extension building. lmk if you'd like to collab here

**Pull request merged**